ta4j-core/src/test/java/org/ta4j/core/indicators/averages/VWMAIndicatorTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2024 Ta4j Organization & respective
+ * authors (see AUTHORS)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.averages;
+
+import static org.ta4j.core.TestUtils.*;
+
+import org.junit.Test;
+import org.ta4j.core.BarSeries;
+import org.ta4j.core.CsvTestUtils;
+import org.ta4j.core.Indicator;
+import org.ta4j.core.indicators.AbstractIndicatorTest;
+import org.ta4j.core.indicators.helpers.ClosePriceIndicator;
+import org.ta4j.core.mocks.MockIndicator;
+import org.ta4j.core.num.Num;
+import org.ta4j.core.num.NumFactory;
+
+public class VWMAIndicatorTest extends AbstractIndicatorTest<Indicator<Num>, Num> {
+
+    public VWMAIndicatorTest(NumFactory numFactory) {
+        super(numFactory);
+    }
+
+    @Test
+    public void vwmaIndicatorTest() {
+
+        MockIndicator mock = CsvTestUtils.getCsvFile(VWMAIndicatorTest.class, "VWMA.csv", numFactory);
```

Comment:
Please flesh out these unit tests further. You can use the CsvTestUtils however they should be in addition to (not a replacement for) atomic test cases including edge conditions. 

When testing using mock data, there should be at minimum one unit test that demonstrates the calculation using simple numbers so that other users can easily reason about the indicator's logic at a glance (as opposed to digesting an entire CSVs worth context).

Please see `RecentSwingLowIndicatorTest `or `KalmanFilterIndicatorTest` as examples of how I expect indicator unit tests to be structured.